mozilla/fx-private-relay
# start new phones page behind waffle flag

This PR fixes MPP-1891.

This starts the "back-end" code for adding an "Upgrade" link. ...
1. Adds a new `PHONE_PROD_ID` environment variable for the Stripe product ID for the new product which grants the new `relay-phones` capability.
2. Adds new `WAFFLE_FLAGS`, `WAFFLE_SWITCHES`, and `WAFFLE_SAMPLES` arrays to the `/api/v1/runtime_data` endpoint.
3. Adds a new `flagIsActive` React function so React code can easily control UI components per flag
   * Note: By using `runtimeData`, `flagIsActive()` will return `true|false` properly even for flags that are per-user, per-group, or %-based flags
4. Adds a new `/phone` route and page, which only shows up if `flagIsActive()` returns `true` for the user



How to test:
1. `git submodule update --init` to get the `privaterelay/locales` branch with the necessary message IDs in `en/app.ftl`
1. `python manage.py waffle_flag phones --create --append --user <your-test-user@example.com>` to create the `phones` waffle flag and activate it only for the `your-test-user@example.com` user
2. Sign in as `your-test-user@example.com`
   * [ ] You should see a `Phone Number` navigation item in the top
4. Click the `Phone Number` navigation item
   * [ ] You should see an `Upgrade now` button on the phone number page
5. Click the `Upgrade now` button
   * [ ] You should see a new product: "Relay fonez (127.0.0.1)"
7. Go thru the FXA stage sub-plat flow
   * [ ] You should be re-directed back to your local Relay site
